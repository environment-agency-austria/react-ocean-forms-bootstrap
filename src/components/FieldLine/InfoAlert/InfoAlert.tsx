/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import * as React from 'react';

import { FormText } from 'react-ocean-forms';
import { Alert } from 'reactstrap';

import { IInfoAlertProps } from './InfoAlert.types';

/**
 * Displays an bootstrap alert containing the parsed
 * info message
 */
export class InfoAlert extends React.Component<IInfoAlertProps> {
  public static displayName: string = 'InfoAlert';

  public render(): JSX.Element | null {
    const {
      visible,
      info,
      plaintext,
      onClose,
    } = this.props;

    if (plaintext || info === undefined) { return null; }

    return (
      <Alert
        color="success"
        className="mt-2"
        isOpen={visible}
        toggle={onClose}
      >
        <FormText text={info} />
      </Alert>
    );
  }
}
