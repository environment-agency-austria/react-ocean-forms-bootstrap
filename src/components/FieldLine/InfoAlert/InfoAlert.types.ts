/**
 * Props for the InfoAlert component
 */
export interface IInfoAlertProps {
  /**
   * True if the info alert should be visible
   */
  visible: boolean;
  /**
   * Info text
   */
  info?: string;
  /**
   * True if the form is in plaintext mode
   */
  plaintext: boolean;
  /**
   * onClose callback
   */
  onClose(): void;
}
