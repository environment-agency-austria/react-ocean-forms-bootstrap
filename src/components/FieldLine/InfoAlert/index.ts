export * from './InfoAlert';
export * from './InfoAlert.types';
