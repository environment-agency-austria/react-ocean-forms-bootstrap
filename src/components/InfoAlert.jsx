/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
import React from 'react';
import PropTypes from 'prop-types';
import { Alert } from 'reactstrap';
import { fieldMetaShape } from 'react-ocean-forms';

/**
 * Displays an bootstrap alert containing the parsed
 * info message
 */
function InfoAlert({
  visible,
  info,
  meta,
  onClose,
}) {
  if (meta.plaintext) return null;
  if (!info) return null;

  const infoString = meta.stringFormatter(info);
  return (
    <Alert
      color="success"
      className="mt-2"
      isOpen={visible}
      toggle={onClose}
    >
      {infoString}
    </Alert>
  );
}

InfoAlert.defaultProps = {
  info: undefined,
};

InfoAlert.propTypes = {
  visible: PropTypes.bool.isRequired,
  info: PropTypes.string,
  meta: fieldMetaShape.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default InfoAlert;
