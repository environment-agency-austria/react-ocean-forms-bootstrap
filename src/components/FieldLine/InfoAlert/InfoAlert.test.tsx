import * as React from 'react';

import { shallow, ShallowWrapper } from 'enzyme';
import { FormText } from 'react-ocean-forms';

import { InfoAlert } from './InfoAlert';

describe('<InfoAlert />', () => {
  const setup = (
    plaintext: boolean,
    info?: string,
    visible: boolean = false,
    onClose: (() => void) = (): undefined => undefined,
  ): ShallowWrapper => {
    return shallow((
      <InfoAlert
        plaintext={plaintext}
        visible={visible}
        info={info}
        onClose={onClose}
      />
    ));
  };

  it('should render nothing if meta.plaintext is active', () => {
    const wrapper = setup(true);
    expect(wrapper.find('Alert').exists()).toBeFalsy();
  });

  it('should render nothing if there is no info', () => {
    const wrapper = setup(false);
    expect(wrapper.find('Alert').exists()).toBeFalsy();
  });

  describe('info present', () => {
    const mockInfo = 'mock-info';
    const mockVisible = true;
    const mockOnClose = jest.fn();

    const wrapper = setup(false, mockInfo, mockVisible, mockOnClose);

    it('should wrap the info with a FormText', () => {
      const formText = wrapper.find(FormText);
      expect(formText).toHaveLength(1);
      expect(formText.prop('text')).toBe(mockInfo);
    });

    it('should render an Alert with the correct props', () => {
      expect(wrapper.find('Alert').exists()).toBeTruthy();

      const alert = wrapper.find('Alert');
      expect(alert.prop('isOpen')).toBe(mockVisible);
      expect(alert.prop('toggle')).toBe(mockOnClose);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
