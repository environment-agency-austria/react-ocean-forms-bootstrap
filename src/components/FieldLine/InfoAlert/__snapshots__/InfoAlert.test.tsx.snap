// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoAlert /> info present should render correctly 1`] = `
<Alert
  className="mt-2"
  closeAriaLabel="Close"
  color="success"
  fade={true}
  isOpen={true}
  tag="div"
  toggle={[MockFunction]}
  transition={
    Object {
      "appear": true,
      "baseClass": "fade",
      "baseClassActive": "show",
      "enter": true,
      "exit": true,
      "in": true,
      "mountOnEnter": false,
      "onEnter": [Function],
      "onEntered": [Function],
      "onEntering": [Function],
      "onExit": [Function],
      "onExited": [Function],
      "onExiting": [Function],
      "tag": "div",
      "timeout": 150,
      "unmountOnExit": true,
    }
  }
>
  <FormText
    text="mock-info"
  />
</Alert>
`;
